core/src/dird/reload.cc

```diff
@@ -48,8 +47,8 @@ bool CheckResources()
          _("No Director resource defined in %s\n"
            "Without that I don't know who I am :-(\n"),
          configfile.c_str());
-    OK = false;
-    goto bail_out;
+    UnlockRes(my_config);
```

Comment:
if we had GSL, you could set this up using `finally()` or if `ConfigurationParser` would satisfy _BasicLockable_ one could just use  a `lock_guard lock{my_config}`. This would remove the need for the explicit `UnlockRes(my_config)`

Reply:
We can probably keep the old behaviour, but allow newer and refactored code to utilize RAII here.

Besides that, you can always do it like this:
```
void afunction() {
  {
     std::lock_guard lock(my_config);
     do_something
  }
  do_something_else
  {
     std::lock_guard lock(my_config);
     doxxx
  }
}
```
which isn't nice, but still a lot better than requiring manual unlock (which is btw. one of the main reasons a lot of our code is not exception-safe)